### PR TITLE
added check to validate password

### LINKED ISSUE #644: Standard regex based password checks
## Description
As a user, we can add certain constraints on the passwords (Combination of numbers/alphabets, not all numbers etc). I don’t think it’s needed presently, but it would be needed as the system scales and it is a feature worth adding, simply implementable by regular expressions. The criteria is taken from Microsoft technet for setting strong passwords.

### Updates required
- [ ] Error message when criteria of password is not matched by the below standard.
- [ ] Passwords must use at least three of the four available character types: lowercase letters, uppercase letters, numbers, and symbols.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by somyaup**
# Description
Passwords must use at least three of the four available character types: lowercase letters, uppercase letters, numbers, and symbols.

Fixes #644 

# Mocks
screenshot from 2018-05-31 22-40-24

# Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

# Checklist:
**Delete irrelevant options.**

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by somyaup** [comment]
@kamsuri @m-murad @ramitsawhney27 @anjali-dhanuka @abhishekarya286 @ayushgarg1804  . Please can you review this ?

File: vms/registration/password_validate.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	if re.search(r'[A-Z]',password) and re.search(r'[0-9]',password) and re.search(r'[@#$%^&+=]',password):
+		upp_num_spec = True
+	if re.search(r'[a-z]',password) and re.search(r'[0-9]',password) and re.search(r'[@#$%^&+=]',password):
+		low_num_spec = True
+
+
+	if upp_num_spec or upp_low_num or upp_low_spec or low_num_spec :
+		return True
+	else:
+		return False
```
Comment (abhishekarya286) [inline vms/registration/password_validate.py]: New line at the end.

**Comment by abhishekarya286** [comment]
Thanks for your contribution @somyaup.

File: vms/registration/password_validate.py (hunk 1 above)
Comment (somyaup) [reply vms/registration/password_validate.py]: done

**Comment by somyaup** [comment]
@ramitsawhney27  Your review turned stale due to my recent commit. Please can you review it again ?

**Comment by sammy1997** [comment]
@anjali-dhanuka @abhishekarya286 LGTM

File: vms/registration/password_validate.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	upp_low_num = False
+	upp_num_spec = False
+	low_num_spec = False
+	if re.search(r'[A-Z]',password) and re.search(r'[a-z]',password) and re.search(r'[@#$%^&+=]',password):
+		upp_low_spec = True
+	if re.search(r'[A-Z]',password) and re.search(r'[a-z]',password) and re.search(r'[0-9]',password):
+		upp_low_num = True
+	if re.search(r'[A-Z]',password) and re.search(r'[0-9]',password) and re.search(r'[@#$%^&+=]',password):
+		upp_num_spec = True
+	if re.search(r'[a-z]',password) and re.search(r'[0-9]',password) and re.search(r'[@#$%^&+=]',password):
```
Comment (ayushgarg1804) [inline vms/registration/password_validate.py]: redundant regex checks in this function. Change it to avoid wasting computations.

**Comment by anjali-dhanuka** [comment]
@SomyaU Work on the merge conflicts and write tests

**Comment by anjali-dhanuka** [comment]
@somyaup Closing your PR. reopen if u plan to work in future.

**Pull request closed without merging**